Core/include/Acts/TrackFitting/GainMatrixSmoother.hpp

```diff
@@ -101,15 +101,15 @@ class GainMatrixSmoother {
       // covariances.
       assert(ts.hasFiltered());
       assert(ts.hasPredicted());
-      assert(ts.hasJacobian());
 
       // previous trackstate should have smoothed and predicted
       assert(prev_ts.hasSmoothed());
       assert(prev_ts.hasPredicted());
+      assert(prev_ts.hasJacobian());
 
       ACTS_VERBOSE("Calculate smoothing matrix:");
       ACTS_VERBOSE("Filtered covariance:\n" << ts.filteredCovariance());
-      ACTS_VERBOSE("Jacobian:\n" << ts.jacobian());
```

Comment:
But we're *using* the correct jacobian, right?

Reply:
yes I tried to follow that through on paper and it looks correct in the source file. But printing here and source file did not line up